src/AddIns/Uno.UI.MediaPlayer.Skia.Gtk/GTKMediaPlayer.cs

```diff
@@ -0,0 +1,240 @@
+#nullable enable
+
+using Windows.UI.Core;
+using LibVLCSharp.GTK;
+using LibVLCSharp.Shared;
+using Windows.UI;
+using Windows.UI.Xaml.Controls;
+using Microsoft.UI.Xaml;
+using System;
+using Windows.UI.Xaml;
+using System.Threading;
+using System.Linq;
+using Windows.UI.Notifications;
+using System.Globalization;
+using System.Collections.Immutable;
+using System.IO;
+using Uno.Extensions;
+
+namespace Uno.UI.Media;
+
+public partial class GTKMediaPlayer : Button
+{
+	private LibVLC? _libvlc;
+	private LibVLCSharp.Shared.MediaPlayer? _mediaPlayer;
+	private VideoView? _videoView;
+	private double _ratio;
+	public double Duration { get; set; }
+	private readonly ImmutableArray<string> audioTagAllowedFormats = ImmutableArray.Create(new string[] { ".MP3", ".WAV" });
+	private readonly ImmutableArray<string> videoTagAllowedFormats = ImmutableArray.Create(new string[] { ".MP4", ".WEBM", ".OGG" });
+	public VideoView? VideoView
+	{
+		get => _videoView;
+		set
+		{
+			_videoView = value;
+		}
+	}
+	public double Ratio
+	{
+		get => _ratio;
+		set
+		{
+			_ratio = value;
+		}
+	}
+	public bool IsVideo
+	{
+		get => videoTagAllowedFormats.Contains(Path.GetExtension(Source), StringComparer.OrdinalIgnoreCase);
+	}
+	public bool IsAudio
+	{
+		get => audioTagAllowedFormats.Contains(Path.GetExtension(Source), StringComparer.OrdinalIgnoreCase);
+	}
+	public GTKMediaPlayer()
+	{
+		//if (this.Log().IsEnabled(LogLevel.Debug))
+		//{
+		//	this.Log().Debug("Adding media elements");
+		//}
+
+		Loaded += OnLoaded;
+		Unloaded += OnUnloaded;
+		//RaiseLoaded();
+		Initialized();
+	}
+
+
+	public void Play()
+	{
+		if (_videoView != null && _mediaPlayer != null)
+		{
+			_mediaPlayer.Play();
+			_videoView.Visible = true;
+		}
+	}
+
+	public void Stop()
+	{
+		if (_videoView != null && _mediaPlayer != null)
+		{
+			_mediaPlayer.Stop();
+			_videoView.Visible = false;
+		}
+	}
+
+	public void Pause()
+	{
+		if (_videoView != null && _mediaPlayer != null)
+		{
+			_mediaPlayer.Pause();
+			_videoView.Visible = true;
+		}
+	}
+
+	public void SetVolume(int volume)
+	{
+		if (_videoView != null && _mediaPlayer != null)
+		{
+			_mediaPlayer.Volume = volume;
+			_videoView.Visible = true;
+		}
+	}
+
+	public string Source
+	{
+		get => (string)GetValue(SourceProperty);
+		set
+		{
+			SetValue(SourceProperty, value);
+		}
+	}
+	public static DependencyProperty SourceProperty { get; } = DependencyProperty.Register(
+		"Source", typeof(string), typeof(GTKMediaPlayer), new PropertyMetadata(default(string),
```

Comment:
Should we use `FrameworkPropertyMetadata` here?

Reply:
We just need a basic metadata relating to the property. 
Should i change it?

Reply:
@jeromelaban knows better :)

Reply:
It don't think it's necessary here. It's only for dependencyobject typed properties. We may end up moving this code out of the uno repo, and FrameworkPropertyMetadata is not available to user code.

Reply:
ok.. Close